tests/testpathgradientbrush.c

```diff
@@ -0,0 +1,1685 @@
+
+    count = 1;
+    status = GdipGetPathGradientSurroundColorsWithCount (brush, colors, &count);
+    assert (status == InvalidParameter);
+
+    count = 0;
+    status = GdipGetPathGradientSurroundColorsWithCount (brush, colors, &count);
+    assert (status == InvalidParameter);
+
+    count = -1;
+    status = GdipGetPathGradientSurroundColorsWithCount (brush, colors, &count);
+    assert (status == InvalidParameter);
+
+    GdipDeleteBrush ((GpBrush *) brush);
+}
+
+static void test_setPathGradientSurroundColorsWithCount ()
+{
+    GpStatus status;
+    GpPathGradient *brush;
+    ARGB threeSurroundColors[3] = {1, 2, 3};
+    ARGB threeEmptyColors[3] = {0, 0, 0};
+    ARGB twoSameColors[2] = {1, 1};
+    ARGB colors[3];
+    INT count = 3;
+
+    GdipCreatePathGradient (threePoints, 3, WrapModeTileFlipX, &brush);
+
+    // Three surround colors.
+    status = GdipSetPathGradientSurroundColorsWithCount (brush, threeSurroundColors, &count);
+    assert (status == Ok);
+    assert (count == 3);
+
+    status = GdipGetPathGradientSurroundColorsWithCount (brush, colors, &count);
+    assert (status == Ok);
+    assert (count == 3);
+    assert (colors[0] == 1);
+    assert (colors[1] == 2);
+    assert (colors[2] == 3);
+
+    // Two empty colors.
+    count = 2;
+    status = GdipSetPathGradientSurroundColorsWithCount (brush, threeEmptyColors, &count);
+    assert (status == Ok);
+    assert (count == 2);
+
+    count = 3;
+    status = GdipGetPathGradientSurroundColorsWithCount (brush, colors, &count);
+    assert (status == Ok);
+    assert (count == 1);
+    assert (colors[0] == 0);
+    assert (colors[1] == 0);
+    assert (colors[2] == 0);
+
+    // Two same colors.
+    count = 2;
+    status = GdipSetPathGradientSurroundColorsWithCount (brush, twoSameColors, &count);
+    assert (status == Ok);
+    assert (count == 2);
+
+    count = 3;
+    status = GdipGetPathGradientSurroundColorsWithCount (brush, colors, &count);
+    assert (status == Ok);
+    assert (count == 1);
+    assert (colors[0] == 1);
+    assert (colors[1] == 1);
+    assert (colors[2] == 1);
+
+    // One surround color.
+    GdipCreatePathGradient (threePoints, 3, WrapModeTileFlipX, &brush);
+
+    count = 1;
+    status = GdipSetPathGradientSurroundColorsWithCount (brush, threeSurroundColors, &count);
+    assert (status == Ok);
+
+    count = 3;
+    status = GdipGetPathGradientSurroundColorsWithCount (brush, colors, &count);
+    assert (status == Ok);
+    assert (count == 1);
+    assert (colors[0] == 1);
+    assert (colors[1] == 1);
+    assert (colors[2] == 1);
+
+    // Negative tests
+    status = GdipSetPathGradientSurroundColorsWithCount (NULL, colors, &count);
+    assert (status == InvalidParameter);
+
+    status = GdipSetPathGradientSurroundColorsWithCount (brush, NULL, &count);
+    assert (status == InvalidParameter);
+
+    // FIXME: this causes GDI+ to crash.
+    // status = GdipSetPathGradientSurroundColorsWithCount (brush, colors, NULL);
```

Comment:
Also a bug in GDI+. Libgdiplus handles this gracefully which I assume is better than actually crashing for compat